proposal/zivzone/proposal.rst

```diff
@@ -0,0 +1,131 @@
+Foe example, there are two common layers in the network. One is pooling and the other is activation function.
+If we change the order of these two layer, the result will be the same but the computation speed will increase.   
+Unfortunately most popular library cannot do this or lacks of freedom to adjust, which means it's hard for me to do the related research by using those common library. 
+So that why I need to build my own library which aims to customize everything and is convineint for me.
+I will start to build the library which can let me build the common DL network first, then I will add the powerful functions like optimizing the architecture if I have enough time.
+
+
+Perspective users
+=================
+People who is working on deep learning can use my library to build thier network easily and use the build-in function to analize the bottleneck of computation and optimize the structure of the network.
+
+
+System architecture
+===================
+The whole library contains different functions in several categories listed below:
+- Activate Fuction
+	- These functions provide the common activate functions for build the network, like Relu and Sigmoid
+- Layers
+	- These functions are used to build the layer in the network, like fully connected layer.  
+- Loss Fuction
+	- We define several common loss function here for evaluate the accuracy of the network, like MSE and L1 Loss.
+- Optimization
+	- We provide several common optimizer for training the network, like SGD.
+- Method
+	- The methods contain the function you can load dataset and autograd for calculate gradient. 
+- Data processing
+	- For data preprocessing
+- Deploy
+	- For deploy the network on different platform.
+- Format/Storage/Data Utilize
+	- Data format transform.
+- Model
+	- Several built-in networks for testing.
+- Case/Application
+	- Some application for example.
+- Visualize
+	- Visualize the result of training and testing.
+- Analyze
+	- Analyze the network like running time, space usage, and flpos counting.
+- Optimize
+	- Provide several method for optimize the model.
+	
+For the user, you may need to define and build the network by using **Layers** and **Activate Fuction**, then loading the dataset and do the data preprocessing by **Method** and **Data processing**.
+Then define your loss function by **Loss Fuction** and **Optimization** for training your network.
+Final, use **Visualize** visualize your training result.
+Additionaly, you can use **Analyze** for analizing your network and use **Optimize** for optimizing your network.
+
+
+API description
+===============
+
+Show how a user programmatically uses your system.  You are supposed to
+implement the system using both C++ and Python.  Describe how a user writes a
+script in the system.  If you provide API in both C++ and Python, describe
+both.
+
+For example, if the user want to build a network (a simple CNN model for classification), they can take the code below as reference:
+```
```

Comment:
This is not markdown.  rst literal block is https://docutils.sourceforge.io/docs/user/rst/quickref.html#literal-blocks .